### DISCUSSION TITLE
make deb fails on raspberry pi 4

### DISCUSSION THREAD

**Discussion by xevilstar** [baresip/rem#70] [Q&A]
src/aubuf/aubuf.c:21:2: error: unknown type name ‘mtx_t’
   21 |  mtx_t *lock;
      |  ^~~~~
src/aubuf/aubuf.c: In function ‘aubuf_alloc’:
src/aubuf/aubuf.c:128:8: warning: implicit declaration of function ‘mtx_alloc’; did you mean ‘rtp_alloc’? [-Wimplicit-function-declaration]
  128 |  err = mtx_alloc(&ab->lock);
      |        ^~~~~~~~~
      |        rtp_alloc
src/aubuf/aubuf.c:128:8: warning: nested extern declaration of ‘mtx_alloc’ [-Wnested-externs]
src/aubuf/aubuf.c: In function ‘aubuf_resize’:
src/aubuf/aubuf.c:184:2: warning: implicit declaration of function ‘mtx_lock’ [-Wimplicit-function-declaration]
  184 |  mtx_lock(ab->lock);
      |  ^~~~~~~~
src/aubuf/aubuf.c:184:2: warning: nested extern declaration of ‘mtx_lock’ [-Wnested-externs]
src/aubuf/aubuf.c:187:2: warning: implicit declaration of function ‘mtx_unlock’ [-Wimplicit-function-declaration]
  187 |  mtx_unlock(ab->lock);
      |  ^~~~~~~~~~
src/aubuf/aubuf.c:187:2: warning: nested extern declaration of ‘mtx_unlock’ [-Wnested-externs]
src/aubuf/aubuf.c: In function ‘aubuf_append_auframe’:
src/aubuf/aubuf.c:239:2: warning: implicit declaration of function ‘list_insert_sorted’; did you mean ‘list_insert_after’? [-Wimplicit-function-declaration]
  239 |  list_insert_sorted(&ab->afl, frame_less_equal, NULL, &f->le, f);
      |  ^~~~~~~~~~~~~~~~~~
      |  list_insert_after
src/aubuf/aubuf.c:239:2: warning: nested extern declaration of ‘list_insert_sorted’ [-Wnested-externs]
make[1]: *** [Makefile:167: build-/aubuf/aubuf.o] Error 1
make[1]: Leaving directory '/usr/src/rem'
make: *** [debian/rules:29: build-stamp] Error 2
dpkg-buildpackage: error: debian/rules build subprocess returned exit status 2

**Comment by sreimers** [comment]
Looks like libre is too old. You need always the same latest stable or dev versions:

https://github.com/baresip/baresip/wiki/Install:-Stable-Release#how-to-install-baresip-stable

**Comment by xevilstar** [comment]
making debsd .....

**Comment by xevilstar** [comment]
librem problem solved

**Comment by xevilstar** [comment]
which lib should I go after now, ffmpeg ?


modules/av1/decode.c: In function ‘av1_decode’:
modules/av1/decode.c:150:2: error: unknown type name ‘aom_codec_frame_flags_t’; did you mean ‘aom_codec_frame_buffer_t’?
  150 |  aom_codec_frame_flags_t flags;
      |  ^~~~~~~~~~~~~~~~~~~~~~~
      |  aom_codec_frame_buffer_t
In file included from /usr/include/aom/aom.h:34,
                 from modules/av1/decode.c:12:
modules/av1/decode.c:294:37: error: ‘AOMD_GET_FRAME_FLAGS’ undeclared (first use in this function)
  294 |  res = aom_codec_control(&vds->ctx, AOMD_GET_FRAME_FLAGS, &flags);
      |                                     ^~~~~~~~~~~~~~~~~~~~
modules/av1/decode.c:294:37: note: each undeclared identifier is reported only once for each function it appears in
modules/av1/decode.c:296:15: error: ‘AOM_FRAME_IS_KEY’ undeclared (first use in this function)
  296 |   if (flags & AOM_FRAME_IS_KEY)
      |               ^~~~~~~~~~~~~~~~
make[1]: *** [mk/mod.mk:49: build-arm64/modules/av1/decode.o] Error 1
make[1]: Leaving directory '/usr/src/baresip'
make: *** [debian/rules:28: build-stamp] Error 2
dpkg-buildpackage: error: debian/rules build subprocess returned exit status 2
